### PR TITLE
Add FIPS related configuration data stubs to the default openssl app config file.

### PR DESCRIPTION
**Pull request by slontis**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: apps/openssl.cnf [hunk 1]
```diff
@@ -11,9 +13,31 @@
...
+# List of providers to load
+[provider_sect]
+default = default_sect
+# The fips section name should match the section name inside the
+# included fipsmodule.cnf.
+# fips = fips_sect
+
+[default_sect]
+# activate = 1
+oid_section = new_oids
```
Comment (levitte) [inline apps/openssl.cnf:37]: Wait, what?  This line should remain in the default section.

Comment (slontis) [reply apps/openssl.cnf:37]: For some reason that made some of the tests actually hang..

Comment (slontis) [reply apps/openssl.cnf:37]: I will debug it to see what is going on..

Comment (slontis) [reply apps/openssl.cnf:37]: Not sure why but today it is working.. All I did was a make update and a rebuild.. weird.

**Comment by levitte** [comment]
We have a misunderstanding, it seems, emanating from sudden double use if "default" in configuration terms.

There is such a thing as a "default section", which has nothing to do with the default provider, but is rather anything that comes before the first named section (i.e. the "default section" is usually unnamed in the config file).

Unfortunately, the way you have currently placed the provider related stuff, the following have been moved from the default section to the *default provider* section:

```
# Extra OBJECT IDENTIFIER info:
#oid_file		= $ENV::HOME/.oid
oid_section		= new_oids

# To use this configuration file with the "-extfile" option of the
# "openssl x509" utility, name here the section containing the
# X.509v3 extensions to use:
# extensions		=
# (Alternatively, use a configuration file that has only
# X.509v3 extensions in its main [= default] section.)
```

So, regarding location, the `openssl_conf = openssl_init` line should indeed be somewhere in the default section (i.e. anywhere before the `[ new_oids ]` section, but the rest must be placed after the whole default section (i.e. just before `[ new_oids ]` or anywhere further down the file).

**Comment by levitte** [comment]
Side note: the default configuration section does have a name.  `[ default ]`.  It should be possible to actually name it, although that's something I've never experimented with...

**Comment by slontis** [comment]
config.pod seems to list the following..

 [openssl_init]
 oid_section = oids
 providers = providers
 alg_section = evp_properties
 ssl_conf = ssl_configuration
 engines = engines

Should it have the oid_section in here then? (That was where I got the info from)..

File: apps/openssl-vms.cnf [hunk 2]
```diff
@@ -1,7 +1,9 @@
 #
 # OpenSSL example configuration file.
-# This is mostly being used for generation of certificate requests.
+# See doc/man5/config.pod for more info.
```
Comment (levitte) [inline apps/openssl-vms.cnf:3]: Oh!  You should probably refer to `openssl-fipsinstall(1)` in the SEE ALSO in `doc/man5/config.pod`

File: apps/openssl.cnf [hunk 3]
```diff
@@ -1,7 +1,9 @@
 #
 # OpenSSL example configuration file.
-# This is mostly being used for generation of certificate requests.
+# See doc/man5/config.pod for more info.
 #
+# This is mostly being used for generation of certificate requests,
```
Comment (richsalz) [inline apps/openssl.cnf:5]: Is this really true?  Lots of TS stuff, for example.

Comment (slontis) [reply apps/openssl.cnf:5]: I am open to suggestions.. tsa is mainly certificate related too right?

File: apps/openssl-vms.cnf (hunk 2 above)
Comment (slontis) [reply apps/openssl-vms.cnf:3]: I added that in this PR..

File: apps/openssl.cnf (hunk 3 above)
Comment (richsalz) [reply apps/openssl.cnf:5]: not a big deal, maybe just delete the comment.  or leave as-is.

Comment (slontis) [reply apps/openssl.cnf:5]: I think I will leave it as is for now.

**Comment by levitte** [comment]
Merged

e45d943665e806ff49d06cfbdd566a8e2d57d56d Add FIPS related configuration data to the default openssl application configuration file

**Pull request closed without merging**